bot/exts/evergreen/wikipedia.py

```diff
@@ -0,0 +1,108 @@
+        async with self.http_session.get(SEARCH_API.format(search_term=search_term)) as response:
+            data = await response.json()
+        page = []
+
+        search_results = data["query"]["search"]
+        if len(search_results) == 0:
+            return None
+
+        # we dont like "may refere to" pages.
+        for search_result in search_results:
+            log.info("trying to appening titles")
+            if "may refer to" in search_result["snippet"]:
+                pass
+            else:
+                page.append(search_result["title"])
+        log.info("Finished appening titles")
+        return page
+
+    @commands.cooldown(1, 10, commands.BucketType.user)
+    @commands.command(name="wikipedia", aliases=["wiki"])
+    async def w_pedia(self, ctx: commands.Context, *, search: str) -> None:
+        """Returns list of your search query from wikipedia."""
+        titles_no_underscore: List[str] = []
+        s_desc = ''
+
+        titles = await self.search_wikipedia(search)
+
+        def check(user: Member) -> bool:
+            return user.author.id == ctx.author.id
+
+        if titles is None:
+            await ctx.send("Sorry, we could not find a wikipedia article using that search term")
+            return
+
+        for title in titles:
+            title_for_creating_link = title.replace(" ", "_")  # wikipedia uses "_" as spaces
+            titles_no_underscore.append(title_for_creating_link)
+        log.info("Finished appening titles with no underscore")
+
+        async with ctx.typing():
+            for index, title in enumerate(titles, start=1):
+                s_desc += f'`{index}` [{title}]({WIKIPEDIA_URL.format(title=title.replace(" ", "_"))})\n'
+            embed = Embed(colour=Color.blue(), title=f"Wikipedia results for `{search}`", description=s_desc)
+            embed.timestamp = datetime.datetime.utcnow()
+            await ctx.send(embed=embed)
+        embed = Embed(colour=Color.green(), description="Enter number to choose")
+        msg = await ctx.send(embed=embed)
+        chances = 0
+        l_of_list = len(titles_no_underscore)  # getting length of list
+
+        while chances <= 3:
+            chances += 1
+            if chances < 3:
+                error_msg = f'You have `{3 - chances}/3` chances left'
+            else:
+                error_msg = 'Please try again by using `.wiki` command'
+            try:
+                user = await ctx.bot.wait_for('message', timeout=60.0, check=check)
+                response = await self.bot.get_context(user)
+                if response.command:
+                    return
+                response = int(user.content)
+                if response < 0:
+                    await ctx.send(f"Sorry, but you can't give negative index, {error_msg}")
+                elif response == 0:
+                    await ctx.send(f"Sorry, please give the range between `1` to `{len(l_of_list)}`, {error_msg}")
```

Comment:
`l_of_list` in this case is an *int* and will raise an exception, as seen here:
```py
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/bot/bot/exts/evergreen/wikipedia.py", line 89, in w_pedia
    await ctx.send(f"Sorry, please give the range between `1` to `{len(l_of_list)}`, {error_msg}")
TypeError: object of type 'int' has no len()
```
This happened when I tried to type `0` as the index.

Reply:
![Screenshot_2020-09-07-23-06-58-857](https://user-images.githubusercontent.com/57266248/92409836-a293be00-f15f-11ea-873f-ed151d077655.jpeg)
I don't know why that happen to you but when u tried that worked fine can u show me what u searched

Reply:
Sure think, I did what you tried as well, and this is what I got:
![image](https://user-images.githubusercontent.com/19921765/92412572-8f401d00-f176-11ea-9386-54dd87bfcee8.png)
![image](https://user-images.githubusercontent.com/19921765/92412579-97985800-f176-11ea-8ab6-a919b8ae6e45.png)
It makes senses as well, since `l_of_list` is indeed an int.